code/clue.js

```diff
@@ -1,59 +1,548 @@
+      // room image
+      document.getElementById("roomImage").src = `${mystery.room.image}`;
+
+      // time delay millisecond
+    }, 300);
+  } /* else {
+    document.getElementById("mystery").innerHTML = "Select a weapon card!";
+  } */
+};
+
+document.getElementById("roomCard").addEventListener("click", pickRoom);
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+// *** Favourite weapon ***
+
+/*
+const shuffleFavouriteWeapon = () => {
+  suspects.forEach((suspect) => {
+    suspect.favouriteWeapon = randomSelector(weapons).name;
+  });
+};
+*/
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+////////////////////////////////////////////////////////////////////////////////////////
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+// *** Reveal Mystery ***
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+const revealMystery = () => {
+  if (mystery.killer === null) {
+    document.getElementById("mystery").innerHTML = "Hey, you have to select a killer...";
+  } else if (mystery.weapon === null) {
+    document.getElementById("mystery").innerHTML = "Select a weapon!";
+  } else if (mystery.room === null) {
+    document.getElementById("mystery").innerHTML = "Select a room!";
+  } else {
+    // "he" or "she"
+    if (
+      mystery.killer.nickName === "Mr Green" ||
+      mystery.killer.nickName === "Professor Plum" ||
+      mystery.killer.nickName === "Colonel Mustard"
+    ) {
+      document.getElementById("mystery").innerHTML = `The murder was committed by the ${mystery.killer.age} years old ${mystery.killer.firstName} ${mystery.killer.lastName} who ${mystery.killer.description}. ${mystery.killer.nickName}'s favourite weapon is the ${mystery.killer.favouriteWeaponId}. ${mystery.killer.lastName} used the ${mystery.weapon.name} to kill the victim in the ${mystery.room.name}. He found the ${mystery.weapon.name} in the ${mystery.weapon.place}.`;
+    } else {
+      document.getElementById("mystery").innerHTML = `The murder was committed by the ${mystery.killer.age} years old ${mystery.killer.firstName} ${mystery.killer.lastName} who ${mystery.killer.description}. ${mystery.killer.nickName}'s favourite weapon is the ${mystery.killer.favouriteWeaponId}. ${mystery.killer.lastName} used the ${mystery.weapon.name} to kill the victim in the ${mystery.room.name}. She found the ${mystery.weapon.name} in the ${mystery.weapon.place}.`;
```

Comment:
Something like this could make it more readable. 

```javascript
let mysteryHTMLContent = "";
mysteryHTMLContent  += `The murder was committed by the ${mystery.killer.age} years old ${mystery.killer.firstName} ${mystery.killer.lastName} who ${mystery.killer.description}. `
mysteryHTMLContent  += `${mystery.killer.nickName}'s favourite weapon is the ${mystery.killer.favouriteWeaponId}. `;
mysteryHTMLContent  += `${mystery.killer.lastName} used the ${mystery.weapon.name} to kill the victim in the ${mystery.room.name}. `
mysteryHTMLContent  += `He found the ${mystery.weapon.name} in the ${mystery.weapon.place}.`;
document.getElementById("mystery").innerHTML = mysteryHTMLContent;
```

How could we do this only once? Maybe something like
```javascript
let genderPronoun = "He";
...
mysteryHTMLContent  += `${genderPronoun} found the ${mystery.weapon.name} in the ${mystery.weapon.place}.`;
document.getElementById("mystery").innerHTML = mysteryHTMLContent;
```